src/envoy/http/jwt_auth/pubkey_cache.h

```diff
@@ -82,6 +82,9 @@ class PubkeyCacheItem : public Logger::Loggable<Logger::Id::filter> {
       if (status != Status::OK) {
         ENVOY_LOG(warn, "Invalid inline jwks for issuer: {}, jwks: {}",
                   jwt_config_.issuer(), inline_jwks);
+        throw EnvoyException(
+            fmt::format("Invalid inline jwks for issuer: {}, jwks: {}",
```

Comment:
I'm trying to test this but it seems this doesn't catch the bad config.

Reply:
Could you add `UpdateStatus(e.GetStatus());` after line 565 in https://github.com/istio/proxy/blob/a975561b980463f08689d3debe33bb9eefc80c3d/src/envoy/http/jwt_auth/jwt.cc#L565

And also add the `StatusToString(status)` to the exception here?

Reply:
I think we actually have validation code to deal with the bad public key but the issue is the result of the validation is not propagated to the caller properly (missing the `UpdateStatus(e.GetStatus());`). If we have this, we would have been able to catch this in very early stage instead of in segfault in runtime.

Reply:
Yes,  I just read the code.  It is very late in throwing exception.  The is constructed in local thread slot.  It is not done in the filter config update.  We should remove the code.  Let me see how easy to add the code in Filter config update.

Reply:
Don't you only just need to call `UpdateStatus(e.GetStatus());` to propagate the status? I thought this would be a easy fix.

Reply:
Done.  add a dummy store in config thread.